### PR TITLE
new package: Lifecycle

### PR DESCRIPTION
**Pull request by thetarnav**
This adds a new `lifecycle` package that extends the lifecycle primitives that solid provides with some new ones.

### `createIsMounted`

```tsx
const isMounted = createIsMounted();
const windowWidth = createMemo(() => (isMounted() ? ref.offsetWidth : 0));

let ref: HTMLElement;
<div ref={ref}>{windowWidth()}</div>;
```

### `isHydrated`

```tsx
// This component will only render its children on the client
export const ClientOnly: FlowComponent = props => {
  const children = createMemo(() => isHydrated() && props.children);
  return <>{children()}</>;
};

// Usage
<ClientOnly>
  <ComponentThatBreaksOnServer />
</ClientOnly>;
```

### `onConnect`

```tsx
<div
  ref={el => {
    // often false, but will be true during hydration
    el.isConnected;

    onMount(() => {
      // often true, but will be false if the executed component is not actually getting rendered
      el.isConnected;
    });

    onConnect(el, () => {
      // always true
      el.isConnected;
    });
  }}
/>
```

[README](https://github.com/solidjs-community/solid-primitives/tree/lifecycle/packages/lifecycle#readme)

### CODE REVIEW

File: packages/lifecycle/README.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+[![turborepo](https://img.shields.io/badge/built%20with-turborepo-cc00ff.svg?style=for-the-badge&logo=turborepo)](https://turborepo.org/)
+[![size](https://img.shields.io/bundlephobia/minzip/@solid-primitives/lifecycle?style=for-the-badge&label=size)](https://bundlephobia.com/package/@solid-primitives/lifecycle)
+[![version](https://img.shields.io/npm/v/@solid-primitives/lifecycle?style=for-the-badge)](https://www.npmjs.com/package/@solid-primitives/lifecycle)
+[![stage](https://img.shields.io/endpoint?style=for-the-badge&url=https%3A%2F%2Fraw.githubusercontent.com%2Fsolidjs-community%2Fsolid-primitives%2Fmain%2Fassets%2Fbadges%2Fstage-0.json)](https://github.com/solidjs-community/solid-primitives#contribution-process)
+
+Package providing extra layer of lifecycle primitives for Solid.
+
+- [`createIsMounted`](#createIsMounted) - Returns a boolean signal indicating whether the component is mounted or not.
+- [`isHydrated`](#isHydrated) - A signal with the same behavior as [`isHydrating`](#isHydrating) but this one focused only on client-side updates.
+- [`onConnect`](#onConnect) - Calls the given callback when the target element is connected to the DOM.
```
Comment (atk) [inline packages/lifecycle/README.md:16]: This name might lead to some confusion with the connectivity package. `onReady` maybe?

**Review by atk — APPROVED** [review]

File: packages/lifecycle/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+ * ```tsx
+ * let ref: HTMLElement
+ * const isMounted = createIsMounted();
+ * const windowWidth = createMemo(() => isMounted() ? ref.offsetWidth : 0)
+ * <div ref={ref}>{windowWidth()}</div>
+ * ```
+ * @see https://github.com/solidjs-community/solid-primitives/tree/main/packages/lifecycle#createIsMounted
+ */
+export function createIsMounted(): Accessor<boolean> {
+  if (isServer) return () => false;
```
[the remark this answers is not in the document]
Comment (thetarnav) [reply packages/lifecycle/src/index.ts:17]: it'll get tree shaken as `isServer` is a const

File: packages/lifecycle/README.md (hunk 1 above)
Comment (thetarnav) [reply packages/lifecycle/README.md:16]: I'm not sure
connect is related to the element being connected to the DOM, if you know about the `el.isConnected` field, you'll understand this one.
if they were in the same package - sure, but they are separated for a reason - component/reactivity lifecycle has nothing to do with internet connection

Comment (thetarnav) [reply packages/lifecycle/README.md:16]: so it's probably the connectivity package that is wrongly named. for "connect" to have meaning it needs to have context

Comment (thetarnav) [reply packages/lifecycle/README.md:16]: I guess I could name it `onElementConnect` to give that missing context here 🤔

Comment (atk) [reply packages/lifecycle/README.md:16]: Maybe `onConnectNode`?

**Pull request merged**